### ISSUE TITLE
GPU tests failing consistently in testgrid

### ISSUE THREAD

**Issue by mboersma** [kubernetes-sigs/cluster-api-provider-azure#1907]
/kind bug

The GPU-enabled cluster tests [have been failing](https://testgrid.k8s.io/sig-cluster-lifecycle-cluster-api-provider-azure#capz-periodic-e2e-full-main) in k8s testgrid. Need to find the root cause.

**Comment by mboersma** [comment]
The driver daemonset is failing to compile GPU support when it starts up:

```
% KUBECONFIG=./e2e.kubeconfig k logs -n gpu-operator-resources nvidia-driver-daemonset-mq7bv                  
Creating directory NVIDIA-Linux-x86_64-460.32.03
Verifying archive integrity... OK
Uncompressing NVIDIA Accelerated Graphics Driver for Linux-x86_64 460.32.03...............................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................

WARNING: Unable to determine the default X library path. The path /tmp/null/lib will be used, but this path was not detected in the ldconfig(8) cache, and no directory exists at this path, so it is likely that libraries installed there will not be found by the loader.


WARNING: You specified the '--no-kernel-module' command line option, nvidia-installer will not install a kernel module as part of this driver installation, and it will not remove existing NVIDIA kernel modules not part of an earlier NVIDIA driver installation.  Please ensure that an NVIDIA kernel module matching this driver version is installed separately.


========== NVIDIA Software Installer ==========

Starting installation of NVIDIA driver version 460.32.03 for Linux kernel version 5.11.0-1021-azure

Stopping NVIDIA persistence daemon...
Unloading NVIDIA driver kernel modules...
Unmounting NVIDIA driver rootfs...
Checking NVIDIA driver packages...
Updating the package cache...
Resolving Linux kernel version...
Proceeding with Linux kernel version 5.11.0-1021-azure
Installing Linux kernel headers...
Installing Linux kernel module files...
Generating Linux kernel version string...
Compiling NVIDIA driver kernel modules...
scripts/Makefile.lib:8: 'always' is deprecated. Please use 'always-y' instead
In file included from /usr/src/nvidia-460.32.03/kernel/nvidia-drm/nvidia-drm-fb.h:38,
                 from /usr/src/nvidia-460.32.03/kernel/nvidia-drm/nvidia-drm-fb.c:29:
/usr/src/nvidia-460.32.03/kernel/nvidia-drm/nvidia-drm-gem-nvkms-memory.h:33:30: error: field 'base' has incomplete type
   33 |     struct nv_drm_gem_object base;
      |                              ^~~~
In file included from <command-line>:
/usr/src/nvidia-460.32.03/kernel/nvidia-drm/nvidia-drm-gem-nvkms-memory.h: In function 'to_nv_nvkms_memory':
./include/linux/kernel.h:694:32: error: dereferencing pointer to incomplete type 'struct nv_drm_gem_object'
  694 |  BUILD_BUG_ON_MSG(!__same_type(*(ptr), ((type *)0)->member) && \
      |                                ^~~~~~
././include/linux/compiler_types.h:300:9: note: in definition of macro '__compiletime_assert'
  300 |   if (!(condition))     \
      |         ^~~~~~~~~
././include/linux/compiler_types.h:320:2: note: in expansion of macro '_compiletime_assert'
  320 |  _compiletime_assert(condition, msg, __compiletime_assert_, __COUNTER__)
      |  ^~~~~~~~~~~~~~~~~~~
./include/linux/build_bug.h:39:37: note: in expansion of macro 'compiletime_assert'
   39 | #define BUILD_BUG_ON_MSG(cond, msg) compiletime_assert(!(cond), msg)
      |                                     ^~~~~~~~~~~~~~~~~~
./include/linux/kernel.h:694:2: note: in expansion of macro 'BUILD_BUG_ON_MSG'
  694 |  BUILD_BUG_ON_MSG(!__same_type(*(ptr), ((type *)0)->member) && \
      |  ^~~~~~~~~~~~~~~~
./include/linux/kernel.h:694:20: note: in expansion of macro '__same_type'
  694 |  BUILD_BUG_ON_MSG(!__same_type(*(ptr), ((type *)0)->member) && \
      |                    ^~~~~~~~~~~
/usr/src/nvidia-460.32.03/kernel/nvidia-drm/nvidia-drm-gem-nvkms-memory.h:47:16: note: in expansion of macro 'container_of'
   47 |         return container_of(nv_gem, struct nv_drm_gem_nvkms_memory, base);
      |                ^~~~~~~~~~~~
/usr/src/nvidia-460.32.03/kernel/nvidia-drm/nvidia-drm-gem-nvkms-memory.h: In function 'to_nv_nvkms_memory_const':
./include/linux/kernel.h:694:32: error: dereferencing pointer to incomplete type 'const struct nv_drm_gem_object'
  694 |  BUILD_BUG_ON_MSG(!__same_type(*(ptr), ((type *)0)->member) && \
      |                                ^~~~~~
././include/linux/compiler_types.h:300:9: note: in definition of macro '__compiletime_assert'
  300 |   if (!(condition))     \
      |         ^~~~~~~~~
././include/linux/compiler_types.h:320:2: note: in expansion of macro '_compiletime_assert'
  320 |  _compiletime_assert(condition, msg, __compiletime_assert_, __COUNTER__)
      |  ^~~~~~~~~~~~~~~~~~~
./include/linux/build_bug.h:39:37: note: in expansion of macro 'compiletime_assert'
   39 | #define BUILD_BUG_ON_MSG(cond, msg) compiletime_assert(!(cond), msg)
      |                                     ^~~~~~~~~~~~~~~~~~
./include/linux/kernel.h:694:2: note: in expansion of macro 'BUILD_BUG_ON_MSG'
  694 |  BUILD_BUG_ON_MSG(!__same_type(*(ptr), ((type *)0)->member) && \
      |  ^~~~~~~~~~~~~~~~
./include/linux/kernel.h:694:20: note: in expansion of macro '__same_type'
  694 |  BUILD_BUG_ON_MSG(!__same_type(*(ptr), ((type *)0)->member) && \
... <hundreds of similar errors omitted>
```

**Comment by mboersma** [comment]
I've updated NVIDIA gpu-operator components to the current versions, and it seems to have fixed this compile problem. I have the tests passing locally, and I'll submit a PR soon.
